v2/internal/reconcilers/reconcile_policy.go

```diff
@@ -17,18 +17,27 @@ import (
 // ParseReconcilePolicy parses the provided reconcile policy.
 // defaultPolicyValue is read from DEFAULT_RECONCILE_POLICY env variable, it set to "manage" when not specified
 func ParseReconcilePolicy(policy string, defaultReconcilePolicy annotations.ReconcilePolicyValue) (annotations.ReconcilePolicyValue, error) {
+	// policy is read from CR annotation, if it's empty it being read from defaultReconcilePolicy
+	reconcilePolicy, err := ParseReconcilePolicyNoDefault(policy)
+	if err != nil {
+		return defaultReconcilePolicy, err
+	}
+	return reconcilePolicy, nil
```

Comment:
Isn't this missing the case where the default policy exists, and the provided policy is `""`?
That shouldn't result in an error it should result in `default, nil`.